### ISSUE TITLE
iOS Analytics Not Showing Up

### ISSUE THREAD

**Issue by RandyCroucher** [firebase/quickstart-unity#101]
I've followed the instructions for both Android and iOS.  Android is working without adding any additional code to my project.  iOS is not.  I am getting a warning in the output that implies I am supposed to add some code, but I'm assuming the Unity plugin is supposed to take care of that for me.

Notable Log Messages...
```
2017-08-16 23:13:59.153169-0700 PCCApp[6525:3234594] [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `[FIRApp configure];` (`FirebaseApp.configure()` in Swift) to your application initialization. Read more: https://goo.gl/ctyzm8.
2017-08-16 23:13:59.153 PCCApp[6525] <Error> [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `[FIRApp configure];` (`FirebaseApp.configure()` in Swift) to your application initialization. Read more: https://goo.gl/ctyzm8.
UnloadTime: 0.234625 ms
2017-08-16 23:14:00.215690-0700 PCCApp[6525:3234418] startMonitoring.
2017-08-16 23:14:00.226582-0700 PCCApp[6525:3234418] No default app present, ignore the previous FIRApp configuration warning (I-COR000003).
No default app present, ignore the previous FIRApp configuration warning (I-COR000003).
Firebase.FirebaseApp:CreateInternal()
Firebase.FirebaseApp:CreateAndTrack(CreateDelegate)
Firebase.Analytics.FirebaseAnalytics:.cctor()
..
2017-08-16 23:14:00.548473-0700 PCCApp[6525:3234595] [Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
2017-08-16 23:14:00.548 PCCApp[6525] <Notice> [Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
2017-08-16 23:14:01.358572-0700 PCCApp[6525:3234562] [Firebase/Analytics][I-ACS032003] iAd framework is not linked. Search Ad Attribution Reporter is disabled.
2017-08-16 23:14:01.358 PCCApp[6525] <Warning> [Firebase/Analytics][I-ACS032003] iAd framework is not linked. Search Ad Attribution Reporter is disabled.
2017-08-16 23:14:01.360785-0700 PCCApp[6525:3234562] [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-16 23:14:01.360 PCCApp[6525] <Notice> [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
```
What am I missing?

**Comment by mcginley561** [comment]
I'm also seeing this issue with the Unity Analytics Test App. Unity 5.5.2f1, Firebase SDK 4.0.3, Xcode 8.3.3 and iOS 10.3.3. As mentioned by @RandyCroucher, there is no issue on Android. iOS Analytics events fail to appear both on the dashboard and the raw export to BigQuery.

Partial log, below. Full log attached
[Firebase4.0.3TestAppiOS-Log.txt](https://github.com/firebase/quickstart-unity/files/1230954/Firebase4.0.3TestAppiOS-Log.txt)

`2017-08-16 19:16:51.150755+0100 education[448:58134] [Firebase/Analytics][I-ACS023068] Data loss. Cannot record event metadata. Error: Error Domain=com.google.gmp.measurement.ErrorDomain Code=2 "Failed to insert/update unknown data type" UserInfo={NSLocalizedDescription=Failed to insert/update unknown data type}
2017-08-16 19:16:51.151 education[448] <Error> [Firebase/Analytics][I-ACS023068] Data loss. Cannot record event metadata. Error: Error Domain=com.google.gmp.measurement.ErrorDomain Code=2 "Failed to insert/update unknown data type" UserInfo={NSLocalizedDescription=Failed to insert/update unknown data type}
2017-08-16 19:16:51.156602+0100 education[448:58127] [Firebase/Analytics][I-ACS034010] Transaction failed
2017-08-16 19:16:51.157 education[448] <Error> [Firebase/Analytics][I-ACS034010] Transaction failed
Set user properties.
UIHandler:DebugLog(String)
UIHandler:InitializeFirebase()
 
(Filename: /Users/builduser/buildslave/unity/build/artifacts/generated/common/runtime/DebugBindings.gen.cpp Line: 51)

2017-08-16 19:16:51.354863+0100 education[448:58135] [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-16 19:16:51.355 education[448] <Notice> [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-16 19:16:51.398235+0100 education[448:58128] [Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
2017-08-16 19:16:51.398 education[448] <Notice> [Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
2017-08-16 19:16:52.464412+0100 education[448:58135] [Firebase/Core][I-COR000020] Error posting to Clearcut: (null), with Status Code: 400
2017-08-16 19:16:52.465 education[448] <Error> [Firebase/Core][I-COR000020] Error posting to Clearcut: (null), with Status Code: 400`

**Comment by stewartmiles** [comment]
Yesterday afternoon we found a bug in Xcode Project level integration of Cocoapods when building for iOS which is probably what both of you are using.  The Firebase iOS 4.0.4 SDK was moved to nanopb which defines a couple of preprocessor macros in the pod (they're not default compile options).  Since our Xcode Project level integration implements a subset of what the Cocoapods "pod" tool does, it ends up *not* applying the required preprocessor macros defined by nanopb.  This results in the Firebase iOS SDK building with one set of options and nanopb in your project (compiled from source) with a different set of options, ultimately a few structures used to call into nanopb differ between the Firebase binary libs and nanopb resulting in mayhem.

So for current users of the 4.0.3 SDK you need to do the following:
* Open their generated Xcode project
* Navigate to Targets -> Unity-iPhone -> Build Settings
* Scroll down to "Preprocessor Macros"
* Add PB_FIELD_32BIT=1 and PB_NO_PACKED_STRUCTS=1 to the list.
* Recompile

We're feverishly working on a fix right now in https://github.com/googlesamples/unity-jar-resolver I'll post an update when we have something you can drop into your projects and then finally when we integrate into the upcoming 4.0.4 release.

**Comment by mcginley561** [comment]
Thanks for that Stewart. I gave it a quick go and it all seems to be working as expected.

**Comment by RandyCroucher** [comment]
Bummer.  Mine already has those entries...

$(inherited) COCOAPODS=1 $(inherited) PB_FIELD_32BIT=1 PB_NO_PACKED_STRUCTS=1

Any other idea?

**Comment by stewartmiles** [comment]
@RandyCroucher hmm I think you may be seeing another issue, could you open an issue with Firebase support (https://firebase.google.com/support/) and they should be able to help diagnose what's going on?

**Comment by stewartmiles** [comment]
@mcginley561 glad it worked out.

Everyone else, I have a temporary patch you can drop into your project.  Replace the existing IOSResolver.dll under PlayServicesResolver/Editor with the attached file and regenerate your Xcode project.

[IOSResolver.dll.zip](https://github.com/firebase/quickstart-unity/files/1232913/IOSResolver.dll.zip)

**Comment by 2yonghyun** [comment]
@stewartmiles Thanks for a solution.
But I hava a question for a temporary patch.
<img width="381" alt="2017-08-20 12 39 04" src="https://user-images.githubusercontent.com/20157363/29491887-dc9b60c8-85a4-11e7-9274-1fee60dc9e5c.png">
In my case, replace Google.IOSResolver_v1.2.35.0.dll  to new Google.IOSResolver.dll you uploaded with changeing a file name equal to old one? Is it right??
And the error message gonna disappear??

**Comment by stewartmiles** [comment]
@2yonghyun, yes you should replace the existing Google.IOSResolver_v1.2.35.0.dll with the new version I shared.

**Comment by stewartmiles** [comment]
The latest version of https://github.com/googlesamples/unity-jar-resolver resolves this issue.  We'll have this integrated into the Firebase Unity SDK shortly.

**Comment by MoinTom** [comment]
I am facing the same issue with FirebaseSDK 4.0.3. Regardless which IOSResolver i am using i get the following:

> 2017-08-22 17:21:17.524739+0200 bloody[634:146815] No default app present, ignore the previous FIRApp configuration warning (I-COR000003).
> No default app present, ignore the previous FIRApp configuration warning (I-COR000003).
> 2017-08-22 17:21:17.528875+0200 bloody[634:147002] [Firebase/Core][I-COR000003] The default Firebase app has not yet been configured. Add `[FIRApp configure];` (`FirebaseApp.configure()` in Swift) to your application initialization. Read more: https://goo.gl/ctyzm8.

I tried the default Resolver, Version 1.2.44.0 as also @stewartmiles one. With Version 1.2.44.0 and @stewartmiles all Preprocessor Macros are defined:

![screen shot 2017-08-22 at 17 25 55](https://user-images.githubusercontent.com/3630717/29573755-126ceb06-8760-11e7-95ac-7da3b6023021.png)


Is there anything else i could try?

**Comment by stewartmiles** [comment]
@MoinTom the couple of warnings you cite in the log are actually spurious, we have suppressed them in our next release of the iOS, C++ and Unity SDKs.  What is the exactly problem you're running into?

**Comment by MoinTom** [comment]
@stewartmiles ah ok good to know. I am implementing FirebaseAnalytics into out app and no data showed up to our console yet. After looking into the log i saw those line and thought this could be a reason for it. Apparently i also see the the following just after the log lines i posted earlier:
```
2017-08-22 17:43:08.646322+0200 bloody[657:150443] [Firebase/Analytics][I-ACS023007] Firebase Analytics v.40002000 started
2017-08-22 17:43:08.646 bloody[657] <Notice> [Firebase/Analytics][I-ACS023007] Firebase Analytics v.40002000 started
2017-08-22 17:43:08.647101+0200 bloody[657:150443] [Firebase/Analytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2017-08-22 17:43:08.648 bloody[657] <Notice> [Firebase/Analytics][I-ACS023008] To enable debug logging set the following application argument: -FIRAnalyticsDebugEnabled (see http://goo.gl/RfcP7r)
2017-08-22 17:43:08.801907+0200 bloody[657:150550] [Firebase/Analytics][I-ACS032003] iAd framework is not linked. Search Ad Attribution Reporter is disabled.
2017-08-22 17:43:08.802 bloody[657] <Warning> [Firebase/Analytics][I-ACS032003] iAd framework is not linked. Search Ad Attribution Reporter is disabled.
2017-08-22 17:43:08.804754+0200 bloody[657:150550] [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-22 17:43:08.806 bloody[657] <Notice> [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-22 17:43:09.181391+0200 bloody[657:150443] [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-22 17:43:09.181 bloody[657] <Notice> [Firebase/Analytics][I-ACS023012] Firebase Analytics enabled
2017-08-22 17:43:09.334587+0200 bloody[657:150406] UnityIAP:Requesting product data...
2017-08-22 17:43:09.473718+0200 bloody[657:150475] [Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
2017-08-22 17:43:09.473 bloody[657] <Notice> [Firebase/Analytics][I-ACS003007] Successfully created Firebase Analytics App Delegate Proxy automatically. To disable the proxy, set the flag FirebaseAppDelegateProxyEnabled to NO in the Info.plist
```
This looks more like a successful initialising to me. But i still don't see any data within my firebase console. Even with *FIRDebugEnabled* and *FIRAnalyticsDebugEnabled* set.

**Comment by stewartmiles** [comment]
@MoinTom the main dashboard does not update in real-time, see 
https://firebase.googleblog.com/2016/11/how-long-does-it-take-for-my-firebase-analytics-data-to-show-up.html

To debug events you can use https://firebase.google.com/docs/analytics/debugview though for the nuance associated with reporting in streamview see https://support.google.com/firebase/answer/7229836?hl=en&ref_topic=6317489

**Comment by MoinTom** [comment]
@stewartmiles i still didn't saw any event within the debugview, but i just noticed that i got a couple of sessions on the streamview. Apparently the use of an updated IOSResolver indeed helped. Thanks for the support.

**Comment by RandyCroucher** [comment]
My analytics are showing up now with the latest build.  Thank you for the help!

**Comment by stewartmiles** [comment]
This issue is now resolved in the 4.1.0 release of the Firebase Unity SDK.  For more information, see
https://firebase.google.com/support/release-notes/unity#4.1.0

Cheers,
Stewart
